### DISCUSSION TITLE
Duty Cycle and Hopping settings for Thread

### DISCUSSION THREAD

**Discussion by nikhil30081995** [openthread/openthread#8226] [Q&A] [answered]
Hello,

We have a product developed using nRF52840 SOC and we are applying for FCC certification.  The product is developed based on nRF5 SDK for Thread v4.1  

I would like to verify few test conditions which are going to be tested during CE and RE testing. 

1. During Transmission mode the duty cycle must be greater than 98%.  So i would like to know whether this condition is satisfied during transmission mode ? 
 
2. The hopping mode should be enabled in the application protocol. So is this mode enabled in Thread ? 

So can anyone please verify the  above  details.

**Comment by jwhui** [comment, marked as answer]
> During Transmission mode the duty cycle must be greater than 98%. So i would like to know whether this condition is satisfied during transmission mode ?

It sounds like you are looking to put the device in continuous transmit mode. If so, this is typically a platform-specific function provided by the radio vendor. I would reach out to Nordic for details.

> The hopping mode should be enabled in the application protocol. So is this mode enabled in Thread ?

Thread does not implement channel hopping.

**Reply by nikhil30081995** [reply]
> > During Transmission mode the duty cycle must be greater than 98%. So i would like to know whether this condition is satisfied during transmission mode ?
> 
> It sounds like you are looking to put the device in continuous transmit mode. If so, this is typically a platform-specific function provided by the radio vendor. I would reach out to Nordic for details.
> 
> > The hopping mode should be enabled in the application protocol. So is this mode enabled in Thread ?
> 
> Thread does not implement channel hopping.

Yeah i have contacted them for the same. 
But i would like to know isn't it the same if I just COAP PUT messages (without ACK) at regular interval of say 100 ms. Because underneath, it will be using IEEE 802.15.4 driver only of nRF52840 SOC.

**Reply by jwhui** [reply]
Simply sending CoAP messages will not reach anywhere near 98% transmit duty-cycle. IEEE 802.15.4 MAC includes CSMA-CA backoffs, synchronous acknowledgments, turn-around times, etc.

**Reply by nikhil30081995** [reply]
> Simply sending CoAP messages will not reach anywhere near 98% transmit duty-cycle. IEEE 802.15.4 MAC includes CSMA-CA backoffs, synchronous acknowledgments, turn-around times, etc.

So if FCC testing requires DC greater than 98% and my end application based on thread won't have DC needed for FCC testing and certification. Than how will i apply for the FCC testing ? 

Also Nordic have suggested me to use 2 examples for testing: 
1 -  Based on OpenThread Diagnostic module, but i don't think it would give me the desired duty cycle of 98% or more using " diag repeat 1 127 " .
2-  Radio test example which basically test the RF at PHY layer. Setting channel, power, modulated tranmission etc. are performed using that example. 

So as my thread based example isn't able to generate duty cycle greater than 98% so if i use Radio test example for certifying my Thread device for FCC standards. Than later on use Thread example on my certified product will it be fine ?

**Reply by jwhui** [reply]
It should be fine to use the radio test example for FCC certification.
